### PR TITLE
Change quality list + make it a required attribute

### PR DESCRIPTION
**Pull request by asaunier**
Fixes https://github.com/c2corg/v6_common/issues/18

### CODE REVIEW

**Comment by tsauerwein** [comment]
You will probably have to adapt the tests in the API, if you make this a required field for all document types.

File: c2corg_common/attributes.py [hunk 1]
```diff
@@ -608,9 +608,9 @@
 ]
 
 quality_types = [
+    'empty',
```
Comment (tsauerwein) [inline c2corg_common/attributes.py:611]: `empty` means what? Isn't this like leaving the field `null` and not using required?

Comment (asaunier) [inline c2corg_common/attributes.py:611]: "empty" is used to tag documents that contains almost no info and are not really interesting for visitors (mainly outings that people have created for their logbook). That's not the default document status (would rather be "medium").

Comment (tsauerwein) [inline c2corg_common/attributes.py:611]: Ok.

**Comment by asaunier** [comment]
Argh...
The API tests return tons of errors. Most likely because I should add this "quality" attribute to all tests since it is supposed to be mandatory...

I wonder if I should not rather:
- leave it not required
- set a default value directly in the field definition in the model

This way if the users do not set a quality, it will at least be set to this default value => same result but simpler, sounds correct?

**Comment by tsauerwein** [comment]
Sounds reasonable. Me as an user I would not know how to rate the quality of my own documents.

**Comment by asaunier** [comment]
OK than I do this way.

I still get tests errors because for instance:

> sqlalchemy.engine.base.Engine: INFO: UPDATE guidebook.documents SET version=%(version)s, quality=%(quality)s WHERE guidebook.documents.document_id = %(guidebook_documents_document_id)s AND guidebook.documents.version = %(guidebook_documents_version)s
> sqlalchemy.engine.base.Engine: INFO: {'quality': None, 'version': 2, 'guidebook_documents_version': 1, 'guidebook_documents_document_id': 1531}
> sqlalchemy.engine.base.Engine: INFO: ROLLBACK
> txn.139844722906880: DEBUG: abort
> c2corg_api: DEBUG: debug_authorization of url http://localhost/waypoints/1531 (view name '' against context IntegrityError('(psycopg2.IntegrityError) null value in column "quality" violates not-null constraint\nDETAIL:  Failing row contains (2, null, null, w, 1531, null).\n',)): Allowed (NO_PERMISSION_REQUIRED)

whereas I have set "quality" in the document model like this:

```
    quality = Column(
        enums.quality_type, nullable=False,
        server_default=quality_types[1], default=quality_types[1])
```

`server_default` changes the table definition:

```
quality      | guidebook.quality_type | not null default 'stub'::guidebook.quality_type
```

Isn't it enough?
I have added `default` as well so that sqlalchemy provides a replacement value if `quality` is not set but it seems to have no effect!?

**Comment by tsauerwein** [comment]
It looks like something is setting `quality` explicitly to null. Is this failing in an update request? You would have to add the `quality` attribute to the request, otherwise `quality` will be None.

**Comment by asaunier** [comment]
I don't know if all tests errors are related to update requests but yes it seems the case of the error quoted above.

> You would have to add the quality attribute to the request, otherwise quality will be None.

Do you mean I should make sure the `quality` attribute is actually set in https://github.com/c2corg/v6_api/blob/master/c2corg_api/models/document.py#L111 ?

**Comment by asaunier** [comment]
Oops, I have forgotten to rebase the fixup commit before merging the PR. Too bad :P

**Pull request merged**